### PR TITLE
Add react-a11y-image-button-has-alt rule

### PR DESCRIPTION
**Pull request by t-ligu**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,6 +134,7 @@ Rule Name   | Description | Since
...
 `promise-must-complete`         | When a Promise instance is created, then either the reject() or resolve() parameter must be called on it within all code branches in the scope. For more examples see the [feature request](https://github.com/Microsoft/tslint-microsoft-contrib/issues/34). | 1.0
 `react-iframe-missing-sandbox`  | React iframes must specify a sandbox attribute. If specified as an empty string, this attribute enables extra restrictions on the content that can appear in the inline frame. The value of the attribute can either be an empty string (all the restrictions are applied), or a space-separated list of tokens that lift particular restrictions. You many not use both allow-scripts and allow-same-origin at the same time, as that allows the embedded document to programmatically remove the sandbox attribute in some scenarios. | 2.0.10
+`react-a11y-image-button-has-alt` | For accessibility of your website, enforce that inputs element of `type="image"` must have an alt attribute. | 2.0.11
```
Comment (ipip2005) [inline README.md]: with

File: src/reactA11yImageButtonHasAltRule.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import * as ts from 'typescript';
+import * as Lint from 'tslint/lib/lint';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+import { getJsxAttributesFromJsxElement, getStringLiteral, isEmpty } from './utils/JsxAttribute';
+
+const NO_ALT_ATTRIBUTE_FAILURE_STRING: string =
+    'Inputs element of type=[image] must have an alt attribute.';
+const EMPTY_ALT_ATTRIBUTE_FAILURE_STRING: string =
+    'Inputs element of type=[image] must have an not empty alt attribute.';
```
Comment (ipip2005) [inline src/reactA11yImageButtonHasAltRule.ts]: non-empty

File: src/reactA11yImageButtonHasAltRule.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * Enforce that inputs element of type=[image] must have an alt attribute.
+ */
+import * as ts from 'typescript';
+import * as Lint from 'tslint/lib/lint';
+
+import { ExtendedMetadata } from './utils/ExtendedMetadata';
+import { getJsxAttributesFromJsxElement, getStringLiteral, isEmpty } from './utils/JsxAttribute';
+
+const NO_ALT_ATTRIBUTE_FAILURE_STRING: string =
+    'Inputs element of type=[image] must have an alt attribute.';
```
Comment (ipip2005) [inline src/reactA11yImageButtonHasAltRule.ts]: remove `an`

File: src/reactA11yImageButtonHasAltRule.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+const EMPTY_ALT_ATTRIBUTE_FAILURE_STRING: string =
+    'Inputs element of type=[image] must have an not empty alt attribute.';
+const TYPE_STRING: string = 'type';
+const ALT_STRING: string = 'alt';
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static metadata: ExtendedMetadata = {
+        ruleName: 'react-a11y-image-button-has-alt',
+        type: 'maintainability',
+        description: 'Enforce that inputs element of type=[image] must have an alt attribute.',
```
Comment (ipip2005) [inline src/reactA11yImageButtonHasAltRule.ts]: remove `an`

**Comment by HamletDRC** [comment]
This was an easy one :)

Once you write a few rules then you become very fast at writing new ones, right?

**Pull request closed without merging**